gapic/templates/%namespace/%name_%version/%sub/services/%service/async_client.py.j2

```diff
@@ -325,10 +325,8 @@ class {{ service.async_client_name }}:
         # so it must be constructed via keyword expansion.
         if isinstance(request, dict):
             request = {{ method.input.ident }}(**request)
-        {% if method.flattened_fields %}{# Cross-package req and flattened fields #}
         elif not request:
-            request = {{ method.input.ident }}({% if method.input.ident.package != method.ident.package %}{% for f in method.flattened_fields.values() %}{{ f.name }}={{ f.name }}, {% endfor %}{% endif %})
-        {% endif %}{# Cross-package req and flattened fields #}
+            request = {{ method.input.ident }}({% if method.flattened_fields %}{% for f in method.flattened_fields.values() %}{{ f.name }}={{ f.name }}, {% endfor %}{% endif %})
         {% else %}
```

Comment:
Do we still need this j2 else case here?

Reply:
Yes, the else case is needed for when`{% if method.input.ident.package != method.ident.package %}` is false and also exists in the sync version here

https://github.com/googleapis/gapic-generator-python/blob/23073450ca2b27bf08a8f005682007efb98fbefa/gapic/templates/%25namespace/%25name_%25version/%25sub/services/%25service/_client_macros.j2#L106-L113